src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -742,6 +754,57 @@ public async Task<int> GetMethodIdByToken(SessionId sessionId, int assembly_id,
             return retDebuggerCmdReader.ReadInt32();
         }
 
+        public async Task<int> GetAssemblyIdFromType(SessionId sessionId, int type_id, CancellationToken token)
+        {
+            var commandParams = new MemoryStream();
+            var commandParamsWriter = new MonoBinaryWriter(commandParams);
+            commandParamsWriter.Write(type_id);
+            commandParamsWriter.Write((int) MonoTypeNameFormat.FormatReflection);
+            var retDebuggerCmdReader = await SendDebuggerAgentCommand<CmdType>(sessionId, CmdType.GetInfo, commandParams, token);
+
+            retDebuggerCmdReader.ReadString();
+
+            retDebuggerCmdReader.ReadString();
+
+            string className = retDebuggerCmdReader.ReadString();
+
+            return retDebuggerCmdReader.ReadInt32();
+        }
+
+        public async Task<List<int>> GetGenericTypeFromType(SessionId sessionId, int type_id, CancellationToken token)
+        {
+            var ret = new List<int>();
+            var commandParams = new MemoryStream();
+            var commandParamsWriter = new MonoBinaryWriter(commandParams);
+            commandParamsWriter.Write(type_id);
+            commandParamsWriter.Write((int) MonoTypeNameFormat.FormatReflection);
+            var retDebuggerCmdReader = await SendDebuggerAgentCommand<CmdType>(sessionId, CmdType.GetInfo, commandParams, token);
+
+            retDebuggerCmdReader.ReadString(); //namespace
+            retDebuggerCmdReader.ReadString(); //name
+            retDebuggerCmdReader.ReadString(); //name full
+            retDebuggerCmdReader.ReadInt32(); //assembly_id
+            retDebuggerCmdReader.ReadInt32(); //module_id
+            retDebuggerCmdReader.ReadInt32(); //type_id
+            retDebuggerCmdReader.ReadInt32(); //rank type
+            retDebuggerCmdReader.ReadInt32(); //type token
+            retDebuggerCmdReader.ReadByte(); //rank
+            retDebuggerCmdReader.ReadInt32(); //flags
+            retDebuggerCmdReader.ReadByte();
+            int nested = retDebuggerCmdReader.ReadInt32();
+            for (int i = 0 ; i < nested; i++)
+            {
+                retDebuggerCmdReader.ReadInt32(); //nested type
+            }
+            retDebuggerCmdReader.ReadInt32(); //typeid
```

Comment:
typeid is a unique identifier that represents a class on debugger side.
In this case if it's a generic type definition it will send the typeId of the original class, if it's a  generic instantiation it will send the typeId of the container_class, or will send 0 if it's not any of this cases.
```
if (mono_class_is_gtd (klass))
    buffer_add_typeid (buf, domain, klass);
else if (mono_class_is_ginst (klass))
    buffer_add_typeid (buf, domain, mono_class_get_generic_class (klass)->container_class);
else
    buffer_add_id (buf, 0);
```